tuna/miopen/celery_tuning/tuning.py

```diff
@@ -0,0 +1,390 @@
+  """Check how many celery workers we need"""
+  worker_granularity = None
+  worker_granularity = get_worker_type(library.args)
+  if 'miopen_find_compile' in library.args.fin_steps \
+  or 'miopen_perf_compile' in library.args.fin_steps:
+    worker_granularity = 'worker_per_node'
+  elif 'miopen_find_eval' in library.args.fin_steps or 'miopen_perf_eval' in library.args.fin_steps:
+    worker_granularity = 'worker_per_gpu'
+
+  return worker_granularity
+
+
+def prep_tuning(library):
+  """Prep env for tuning start"""
+  worker_type = get_worker_type(library.args)
+  machines = load_machines(library.args)
+  q_name = None
+  if worker_type == 'fin_build_worker':
+    q_name = f"compile_q_session_{library.dbt.session_id}"
+  else:
+    q_name = f"eval_q_session_{library.dbt.session_id}"
+
+  if not library.args.enqueue_only:
+    try:
+      #stop_active_workers()
+      if not launch_celery_worker(library, q_name, machines,
+                                  get_worker_granularity(library)):
+        raise ValueError('Could not launch celery worker')
+    except kombu.exceptions.OperationalError as k_err:
+      LOGGER.error('Redis error ocurred: %s', k_err)
+      return False
+
+  global DBT  #pylint: disable=global-variable-undefined
+  DBT = MIOpenDBTables(session_id=library.args.session_id,
+                       config_type=library.args.config_type)
+
+  fdb_attr = [column.name for column in inspect(DBT.find_db_table).c]
+  fdb_attr.remove("insert_ts")
+  fdb_attr.remove("update_ts")
+
+  f_vals = library.get_f_vals(Machine(local_machine=True),
+                              range(0),
+                              tuning=True)
+  kwargs = library.get_kwargs(0, f_vals, tuning=True)
+
+  return worker_type, kwargs, fdb_attr, q_name
+
+
+#pylint: disable=too-many-locals
+def tune(library, job_batch_size=1000):
+  """tuning loop to spin out celery tasks"""
+
+  try:
+    worker_type, kwargs, fdb_attr, q_name = prep_tuning(library)
+  except ValueError as verr:
+    LOGGER.error(verr)
+    return False
+
+  #if enqueue_only is False, we only launch the workers
+  if not library.args.enqueue_only:
+    return True
+
+  res_set = ResultSet([])
+  start = time.time()
+
+  with DbSession() as session:
+    while True:
+      try:
+        job_list = []
+        #get all the jobs from mySQL
+        job_list = library.get_jobs(session, library.fetch_state,
+                                    library.set_state, library.args.session_id,
+                                    job_batch_size)
+
+        for i in range(0, len(job_list), job_batch_size):
+          batch_jobs = job_list[i:min(i + job_batch_size, len(job_list))]
+          if library.args.fin_steps:
+            entries = library.compose_work_objs_fin(session, batch_jobs,
+                                                    library.dbt)
+
+          entries = library.compose_work_objs_fin(session, job_list,
+                                                  library.dbt)
```

Comment:
Appears to always overwrite entries. Desired behavior?

Reply:
thanks for catching, not intended behavior